Mathlib/Order/Category/BddLatCat.lean

```diff
@@ -0,0 +1,238 @@
+  unitIso := NatIso.ofComponents fun X => Iso.mk <| OrderIso.dualDual X
+  counitIso := NatIso.ofComponents fun X => Iso.mk <| OrderIso.dualDual X
+#align BddLat.dual_equiv BddLatCat.dualEquiv
+
+end BddLatCat
+
+theorem bddLatCat_dual_comp_forget_to_bddOrdCat :
+    BddLatCat.dual ⋙ forget₂ BddLatCat BddOrdCat =
+    forget₂ BddLatCat BddOrdCat ⋙ BddOrdCat.dual :=
+  rfl
+#align BddLat_dual_comp_forget_to_BddOrd bddLatCat_dual_comp_forget_to_bddOrdCat
+
+theorem bddLatCat_dual_comp_forget_to_latCat :
+    BddLatCat.dual ⋙ forget₂ BddLatCat LatCat = forget₂ BddLatCat LatCat ⋙ LatCat.dual :=
+  rfl
+#align BddLat_dual_comp_forget_to_Lat bddLatCat_dual_comp_forget_to_latCat
+
+theorem bddLatCat_dual_comp_forget_to_semilatSupCat :
+    BddLatCat.dual ⋙ forget₂ BddLatCat SemilatSupCat =
+    forget₂ BddLatCat SemilatInfCat ⋙ SemilatInfCat.dual :=
+  rfl
+#align BddLat_dual_comp_forget_to_SemilatSup bddLatCat_dual_comp_forget_to_semilatSupCat
+
+theorem bddLatCat_dual_comp_forget_to_semilatInfCat :
+    BddLatCat.dual ⋙ forget₂ BddLatCat SemilatInfCat =
+    forget₂ BddLatCat SemilatSupCat ⋙ SemilatSupCat.dual :=
+  rfl
+#align BddLat_dual_comp_forget_to_SemilatInf bddLatCat_dual_comp_forget_to_semilatInfCat
+
+/-- The functor that adds a bottom and a top element to a lattice. This is the free functor. -/
+def latToBddLatCat : LatCat.{u} ⥤ BddLatCat where
+  obj X := BddLatCat.of <| WithTop <| WithBot X
+  map := LatticeHom.withTopWithBot
+  map_id _ := LatticeHom.withTopWithBot_id
+  map_comp _ _ := LatticeHom.withTopWithBot_comp _ _
+#align Lat_to_BddLat latToBddLatCat
+
+/-- `latToBddLatCat` is left adjoint to the forgetful functor, meaning it is the free
+functor from `Lat` to `BddLatCat`. -/
+def latToBddLatCatForgetAdjunction : latToBddLatCat.{u} ⊣ forget₂ BddLatCat LatCat :=
+  Adjunction.mkOfHomEquiv
+    { homEquiv := fun X Y =>
+        { toFun := fun f =>
+            { toFun := f ∘ some ∘ some
+              map_sup' := fun a b => (congr_arg f <| by rfl).trans (f.map_sup' _ _)
+              map_inf' := fun a b => (congr_arg f <| by rfl).trans (f.map_inf' _ _) }
+          invFun := fun f => LatticeHom.withTopWithBot' f
+          left_inv := fun f =>
+            BoundedLatticeHom.ext fun a =>
+              match a with
+              | none => f.map_top'.symm
+              | some none => f.map_bot'.symm
+              | some (some a) => rfl
+          right_inv := fun f => LatticeHom.ext fun a => rfl }
+      homEquiv_naturality_left_symm := fun f g =>
+        BoundedLatticeHom.ext fun a =>
+          match a with
+          | none => rfl
+          | some none => rfl
+          | some (some a) => rfl
+      homEquiv_naturality_right := fun f g => LatticeHom.ext fun a => rfl }
+#align Lat_to_BddLat_forget_adjunction latToBddLatCatForgetAdjunction
+
+
+/-- `latToBddLatCat` and `OrderDual` commute. -/
+-- Porting note: There is a warning plus `simpNF` linter complaining which I don't feel like
+-- debugging at the moment.
+-- @[simps]
```

Comment:
The message is just saying that you need to switch to `@[simps!]`. I've pushed this change.

Reply:
That porting note wasn't that clear: the part I didn't understand was the `simpNF` error:

```
Left-hand side simplifies from
  ↑(latToBddLatCatCompDualIsoDualCompLatToBddLatCat.hom.app X).toSupHom a✝
to
  ↑(latToBddLatCatCompDualIsoDualCompLatToBddLatCat.hom.app X).toSupHom a✝
```

but I'll look into it

Reply:
Nah I got nothing :(

@semorrison would you like to have this investigated or is `nolint simpNF` okay?

Reply:
`nolint simpNF` is not enough. I went back to uncommenting `@[simps!]`. If somebody wants to investigate before this is merged, feel free!